python-connectors/data/connector.py

```diff
@@ -24,12 +25,21 @@ def get_read_schema(self):
 
     def generate_rows(self, dataset_schema=None, dataset_partitioning=None,
                       partition_id=None, records_limit=-1):
+        nb_items = 0
+        maximum_items = records_limit if self.maximum_items == -1 else min([records_limit, self.maximum_items])
```

Comment:
You could have use float('inf') instead of -1 and just take min(records_limit, self.maximum_items) for simplicity